### PR TITLE
ENH: Add qMRI example datasets

### PR DESCRIPTION
**Pull request by effigies**
Adapted examples from https://osf.io/k4bs5/ for BEP001.

For easier review, here are the dataset trees:

* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_mp2rage
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_mp2rageme
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_mpm
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_mtsat
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_qsm
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_sa2rage
* https://github.com/effigies/BIDS-examples/tree/qmri/qmri_vfa

### CODE REVIEW

**Comment by effigies** [comment]
Getting errors about `fmap/*_TB1map.nii`. Should these be moved to `anat/` or should the spec be updated to put them in `fmap/`?

cc @agahkarakuzu @tsalo (please cc anybody else who should be added)

**Comment by tsalo** [comment]
They should be under `fmap` (see [here](https://github.com/bids-standard/bids-specification/blob/7cdeadf6d0f448cbf5fe0d30d75787fb3166bbee/src/schema/datatypes/fmap.yaml#L35-L67)), but I missed them in https://github.com/bids-standard/bids-validator/pull/1161. 🤦

I can open a PR to the validator to fix that oversight.

**Comment by effigies** [comment]
@tsalo To be clear, I meant TB1Map/RB1Map ([schema](https://github.com/bids-standard/bids-specification/blob/7cdeadf6d0f448cbf5fe0d30d75787fb3166bbee/src/schema/datatypes/anat.yaml#L42-L43)). Those are declared in anatomical.

File: qmri_mpm/sub-01/fmap/sub-01_phasediff.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "ManufacturersModelName": "Prisma_fit",
+  "DeviceSerialNumber": "167025",
+  "StationName": "MRC35437",
+  "MagneticFieldStrength": 3,
+  "ScanningSequence": "GR",
+  "SequenceName": "*fm2d2r ",
+  "PulseSequenceDetails": "gre_field_mapping_1acq_rl ",
+  "RepetitionTimeExcitation": 1.02,
+  "EchoTime1": 0.01246,
+  "EchoTime2": 0.01246,
```
Comment (sappelhoff) [inline qmri_mpm/sub-01/fmap/sub-01_phasediff.json:12]: https://github.com/bids-standard/bids-examples/pull/245/checks?check_run_id=1872898390

This seems like an error unrelated to missing qMRI coverage

image

File: qmri_mpm/sub-01/fmap/sub-01_magnitude1.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "MTState": false,
+  "NumberShots": 1,
+  "PhaseEncodingDirection": "j",
+  "IntendedFor": [
+    "fmap/sub-anon_acq-headMTw_RB1COR.nii",
+    "fmap/sub-anon_acq-headPDw_RB1COR.nii",
+    "fmap/sub-anon_acq-headT1w_RB1COR.nii",
+    "fmap/sub-anon_acq-bodyMTw_RB1COR.nii",
+    "fmap/sub-anon_acq-bodyPDw_RB1COR.nii",
+    "fmap/sub-anon_acq-bodyT1w_RB1COR.nii"
```
Comment (sappelhoff) [inline qmri_mpm/sub-01/fmap/sub-01_magnitude1.json:22]: Is there a missing `sub-anon` that I just don't see, or is this an error?

https://github.com/bids-standard/bids-examples/pull/245/checks?check_run_id=1872898390

image

**Comment by effigies** [comment]
Nice catches, @sappelhoff. I think we're down to 

* Pending final decisiions for https://github.com/bids-standard/bids-specification/pull/728 and a corresponding validator PR.
* Clarification on whether `T`/`RB1Map`s should be fieldmap or anat and how to encode that in the spec, and a corresponding validator PR.

File: qmri_mpm/sub-01/fmap/sub-01_echo-1_flip-01_TB1EPI.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "MagneticFieldStrength": 3,
+  "ScanningSequence": "RM",
+  "SequenceName": "seste1d3d2",
+  "PulseSequenceDetails": "mfc_seste_b1map_v1e ",
+  "RepetitionTimeExcitation": 0.5,
+  "EchoTime": 0.03906,
+  "FlipAngle": 65,
+  "MTState": false,
+  "NumberShots": 11,
+  "TotalReadoutTime": 6720.006240000001,
```
Comment (sappelhoff) [inline qmri_mpm/sub-01/fmap/sub-01_echo-1_flip-01_TB1EPI.json:15]: We are getting a warning for the `TotalReadouttime` fields in the `qmri_mpm` datasets:

Seems valid:

image

**Review by sappelhoff — COMMENTED** [review]
we also get warnings about missing Author fields. I inquired with the BEP001 team how to resolve this: https://mattermost.brainhack.org/brainhack/pl/icwy7hqqwtnjzjrzpmmgmehgby

Probably adding something like `["qmri BEP Team", "BIDS maintainers"]` will be fine.

**Comment by Gilles86** [comment]
@sappelhoff ,

The MP2RAGE and MP2RAGE-ME datasets are authored by myself and Tomas Knapen.

For the MPM I think you should check with @ChristophePhillips and @lazaral 

I think the rest has been provided by @agahkarakuzu

**Comment by agahkarakuzu** [comment]
@sappelhoff I noticed that the validator was complaining about single authorship :) I started expanding the example datasets and validator, I can edit them as I go by preserving 1st dataset providers' order.

**Comment by sappelhoff** [comment]
Thanks both, I think it'd be best if you could update the OSF datasets with the authors, and mirror the changes that @effigies already did (see e.g., https://github.com/bids-standard/bids-examples/pull/245/commits/7e1ec5580da2c37e699f626747f4d71ec5fb40ec and https://github.com/bids-standard/bids-examples/pull/245/commits/8d6f42d94670c70486b6d04ced1350e1d1433579) --> Then Chris can pull these changes into his datalad controlled datasets, and then push them to bids-examples.

Perhaps you can also solve https://github.com/bids-standard/bids-examples/pull/245#discussion_r575640061 as you go

as for validator, please see: https://github.com/bids-standard/bids-validator/pull/1186

from that PR, the validator now passes for qMRI.

**Comment by lazaral** [comment]
@sappelhoff, regarding the MPM dataset, we used the one from this publication: https://www.sciencedirect.com/science/article/pii/S235234091930486X?via%3Dihub

We currently have the first author of that manuscript names as author of the dataset in the OSF repo. Should we add the full author list to the dataset_description.json?

**Comment by sappelhoff** [comment]
> Should we add the full author list to the dataset_description.json?

@lazaral yes please, make it an array of strings: `["author1", "author2", ...]`

**Review by sappelhoff — APPROVED** [review]
All datasets are passing bids-validator when validating using the changes from https://github.com/bids-standard/bids-validator/pull/1186


As I mention in https://github.com/bids-standard/bids-validator/pull/1186#issuecomment-779333369, I suggest to merge this PR into the https://github.com/bids-standard/bids-examples/tree/bep001_qmri branch.

Then we can sort out the final issues and them merge that one into `master`

**Pull request merged**